src/routes/signup/+page.server.ts

```diff
@@ -0,0 +1,59 @@
+import { lucia } from "$lib/server/auth";
+import { fail, redirect, type Actions } from "@sveltejs/kit";
+import { generateIdFromEntropySize } from "lucia";
+import { hash } from "@node-rs/argon2";
+
+import db from "$lib/db";
+import { users } from "$lib/db/schema";
+
+export const actions: Actions = {
+	default: async (event) => {
+		const formData = await event.request.formData();
+		const username = formData.get("username");
+		const password = formData.get("password");
+		// username must be between 4 ~ 31 characters, and only consists of lowercase letters, 0-9, -, and _
+		// keep in mind some database (e.g. mysql) are case insensitive
+		if (
+			typeof username !== "string" ||
+			username.length < 3 ||
+			username.length > 31 ||
+			!/^[a-z0-9_-]+$/.test(username)
+		) {
+			return fail(400, {
+				message: "Invalid username"
+			});
+		}
+		if (typeof password !== "string" || password.length < 6 || password.length > 255) {
+			return fail(400, {
+				message: "Invalid password"
+			});
+		}
+
+		const userId = generateIdFromEntropySize(10); // 16 characters long
+		const passwordHash = await hash(password, {
+			// recommended minimum parameters
+			memoryCost: 19456,
+			timeCost: 2,
+			outputLen: 32,
+			parallelism: 1
+		});
+		
+		// TODO: check if username is already used
+		await db.insert(users).values({
+			id: userId,
+			firstname: username,
+            lastname: username,
+            email: username,
```

Comment:
Fix indentation

Reply:
I'm not suer what you mean. If you don't like how it's formatted you can change it.